modules/flowable-engine/src/test/java/org/flowable/engine/test/api/event/TransactionEventListenerTest.java

```diff
@@ -28,63 +28,70 @@
 import org.flowable.engine.test.Deployment;
 
 public class TransactionEventListenerTest extends PluggableFlowableTestCase {
-    
+
     protected TestTransactionEventListener onCommitListener;
-    
+
     @Override
     protected void setUp() throws Exception {
         super.setUp();
-        
+
         onCommitListener = new TestTransactionEventListener(TransactionState.COMMITTED.name());
         processEngineConfiguration.getEventDispatcher().addEventListener(onCommitListener);
     }
-    
+
     @Override
     protected void tearDown() throws Exception {
         TestTransactionEventListener.eventsReceived.clear();
         if (onCommitListener != null) {
             processEngineConfiguration.getEventDispatcher().removeEventListener(onCommitListener);
             onCommitListener = null;
         }
-        
+
         super.tearDown();
     }
-    
+
     public void testRegularProcessExecution() {
-        
+
         assertEquals(0, TestTransactionEventListener.eventsReceived.size());
-        
-        // In a 'normal' process execution, the transaction dependent event listener should 
+
+        // In a 'normal' process execution, the transaction dependent event listener should
         // be similar to the normal event listener dispatching.
-        
+
         deployOneTaskTestProcess();
         runtimeService.startProcessInstanceByKey("oneTaskProcess");
-        
-        assertEquals(9, TestTransactionEventListener.eventsReceived.get(FlowableEngineEventType.ENTITY_CREATED.name()).size());
-        assertEquals(9, TestTransactionEventListener.eventsReceived.get(FlowableEngineEventType.ENTITY_INITIALIZED.name()).size());
+
+        int expectedCreatedEvents = 10;
+        if (processEngineConfiguration.isAsyncHistoryEnabled()) {
+            waitForHistoryJobExecutorToProcessAllJobs(5000L, 100L);
+            // in the case of async history, there is one additional create/initialized event related to newly created HistoryJob
+            expectedCreatedEvents = 11;
```

Comment:
the qa build http://qa.flowable.org/view/Version%206/job/Flowable6MySql_AsyncHistory-martin/1/